proxy/http/HttpSM.cc

```diff
@@ -6837,6 +6847,10 @@ HttpSM::kill_this()
   //   then the value of kill_this_async_done has changed so
   //   we must check it again
   if (kill_this_async_done == true) {
+    if (pending_action) {
```

Comment:
makes sense according to the backtraces.
might remove line 6854 as well since it's useless now.